crates/subgraph/src/types/impls.rs

```diff
@@ -29,3 +97,114 @@ mod js_api {
     impl_wasm_traits!(BigInt);
     impl_wasm_traits!(Bytes);
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::types::common::{
+        BigInt, Bytes, Orderbook, TradeEvent, TradeStructPartialOrder, TradeVaultBalanceChange,
+        Transaction, VaultBalanceChangeVault,
+    };
+    use alloy::primitives::Address;
+
+    #[test]
+    fn test_scale_18_io() {
+        let (input, output) = get_trade().scale_18_io().unwrap();
+        let expected_input = U256::from_str("3000000000000000000").unwrap();
+        let expected_output = U256::from_str("6000000000000000000").unwrap();
+        assert_eq!(input, expected_input);
+        assert_eq!(output, expected_output);
+    }
+
+    #[test]
+    fn test_ratio() {
+        let result = get_trade().ratio().unwrap();
+        let expected = U256::from_str("500000000000000000").unwrap();
+        assert_eq!(result, expected);
+    }
+
+    #[test]
+    fn test_inverse_ratio() {
```

Comment:
you added if cases in the inverse ratio fn and didn't test them

Reply:
added coverage